File: Spring-Working-Time-Management/src/main/java/com/example/miniprojetjee/entity/Work.java
package com.example.miniprojetjee.entity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalTime;
import java.util.Date;
import java.util.List;
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Work {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Date date;
    private LocalTime start;
    private LocalTime end;
    @ManyToOne
    @JoinColumn(name = "employee_id")
    private Employee employee;
    @OneToMany(mappedBy = "work")
    private List<Breaks> breaks;
    private boolean isEmpty;


}

File: Front_GI/src/app/models/work.ts
import { Employee } from "./employee";

export class Work {

    constructor(
        public id: number,
        public date: Date,
        public start: string,
        public end: string,
        public employee?: Employee,
        public isEmpty?: boolean,
        public duration?: number,
        public breakDuration?: number
    ){
        
    }

}